server.go

```diff
@@ -830,7 +830,33 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 	// With the announcement generated, we'll use our getNodeAnnouncement
 	// helper to sign our current announcement to properly authenticate
 	// with the network.
-	nodeAnn, err := s.genNodeAnnouncement(true)
+	var mods []netann.NodeAnnModifier
+	if len(s.cfg.ProtocolOptions.SetFeatureBit()) > 0 {
```

Comment:
Put the result of `SetFeatureBit` in a variable to not call the method twice?

And is this if statement necessary? I'd think that if there are no custom features, then the code below is a no-op. It would reduce the number of code paths.